### DISCUSSION TITLE
Understanding End2End

### DISCUSSION THREAD

**Discussion by knoppmyth** [ultralytics/ultralytics#13262] [General]
Hi All,

I'm trying to get an understanding of end2end and it's benefits. Based on what I'm read, building an end2end model (in yolo's context), means a model that internally includes NMS functionality. This potentially means a speedier model without needing an external NMS step. Is this correct?

Thanks,

km

**Comment by pderrenger** [comment]
@knoppmyth hi km,

You're on the right track! An end-to-end (E2E) model in the context of YOLO indeed aims to integrate functionalities like Non-Maximum Suppression (NMS) directly into the model. This integration can lead to faster inference times as it reduces the need for separate post-processing steps. By handling everything within the model, it streamlines the process, potentially enhancing both speed and efficiency during deployment.

If you have more questions or need further clarification, feel free to ask!
